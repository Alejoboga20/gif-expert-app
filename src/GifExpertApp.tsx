import { useState } from 'react';
import { Text } from '@nextui-org/react';
import { AddCategory } from './components/AddCategory/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }: GifExpertAppProps) => {
	const [categories, setCategories] = useState(defaultCategories);

	return (
		<div>
			<Text h2>Gif Expert App</Text>
			<AddCategory setCategories={setCategories} />
			<hr />
			<ol>
				{categories.map((category) => (
					<GifGrid key={category} category={category} />
				))}
			</ol>
		</div>
	);
};

interface GifExpertAppProps {
	defaultCategories?: string[];
}

export default GifExpertApp;
