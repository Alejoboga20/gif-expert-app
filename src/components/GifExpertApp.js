import React, { useState } from 'react';
import AddCategory from './AddCategory';
import GifGrid from './GifGrid';

const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);

  return (
    <div>
      <h2>Gif Expert App</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </div>
  );
};

export default GifExpertApp;
