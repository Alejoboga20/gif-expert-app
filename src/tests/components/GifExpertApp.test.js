import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../../GifExpertApp';

describe('GifExpertApp Tests', () => {
  let wrappper = shallow(<GifExpertApp />);

  test('should Display GifExpertApp properly', () => {
    expect(wrappper).toMatchSnapshot();
  });

  test('should show a list of categories', () => {
    const categories = ['Avatar', 'Lord Of Rings'];
    wrappper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrappper).toMatchSnapshot();
    expect(wrappper.find('GifGrid').length).toBe(categories.length);
  });
});
