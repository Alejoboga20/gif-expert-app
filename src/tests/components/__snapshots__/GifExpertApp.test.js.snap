// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp Tests should Display GifExpertApp properly 1`] = `
<div>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`GifExpertApp Tests should show a list of categories 1`] = `
<div>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Avatar"
      key="Avatar"
    />
    <GifGrid
      category="Lord Of Rings"
      key="Lord Of Rings"
    />
  </ol>
</div>
`;
